cogboard-webapp/src/components/CredentialForm/index.js

```diff
@@ -38,31 +43,74 @@ const CredentialsForm = ({
   const validationSchema = createValidationSchema(formFields, constraints);
   const { values, handleChange, withValidation, errors } = useFormData(
     initialFormValues,
-    { initialSchema: validationSchema, onChange: true }
+    {
+      initialSchema: validationSchema,
+      onChange: true
+    }
   );
 
+  const [tabValue, setTabValue] = useState(0);
+
+  const handleTabChange = (_, newValue) => {
+    setTabValue(newValue);
+  };
+
   return (
-    <form onSubmit={withValidation(onSubmit)} noValidate="novalidate">
-      <DynamicForm
-        fields={formFields}
-        values={values}
-        handleChange={handleChange}
-        errors={errors}
-        rootName="credential-form"
-      />
-      <Button
-        color="secondary"
-        variant="contained"
-        type="submit"
-        data-cy="credential-form-submit-button"
+    <div>
+      <StyledTabs
+        value={tabValue}
+        onChange={handleTabChange}
+        variant="fullWidth"
+        indicatorColor="primary"
       >
-        Save
-      </Button>
-      <StyledCancelButton
-        handleCancelClick={handleCancel}
-        data-cy="credential-form-cancel-button"
-      />
-    </form>
+        <Tab label="Basic" data-cy="credential-form-basic-tab" />
+        <Tab label="Token" data-cy="credential-form-token-tab" />
+      </StyledTabs>
+      <form onSubmit={withValidation(onSubmit)} noValidate="novalidate">
+        <StyledTabPanel value={tabValue} index={0}>
+          <DynamicForm
+            fields={basicFormFields}
+            values={values}
+            handleChange={handleChange}
+            errors={errors}
+            rootName="credential-form"
+          />
+          <Button
+            color="secondary"
+            variant="contained"
+            type="submit"
+            data-cy="credential-form-submit-button"
+          >
+            Save
+          </Button>
+          <StyledCancelButton
+            handleCancelClick={handleCancel}
+            data-cy="credential-form-cancel-button"
+          />
+        </StyledTabPanel>
+        <StyledTabPanel value={tabValue} index={1}>
+          <DynamicForm
+            fields={tokenFormFields}
+            values={values}
+            handleChange={handleChange}
+            errors={errors}
+            rootName="credential-form"
+          />
+          <Button
+            color="secondary"
+            variant="contained"
+            type="submit"
+            data-cy="credential-form-submit-button"
+          >
+            Save
+          </Button>
+          <StyledCancelButton
+            handleCancelClick={handleCancel}
+            data-cy="credential-form-cancel-button"
+          />
```

Comment:
Can we avoid this duplication? What if you move this outside of Tabs?

Reply:
fixed